diffrax/_solver/srk.py

```diff
@@ -514,11 +520,10 @@ def compute_and_insert_kf_j(_h_kfs_in):
             if ignore_stage_f is None:
                 _h_kfs = compute_and_insert_kf_j(_h_kfs)
             else:
-                drift_pred = jnp.logical_not(ignore_stage_f[j])
                 _h_kfs = lax.cond(
-                    eqxi.nonbatchable(drift_pred),
-                    compute_and_insert_kf_j,
+                    eqxi.nonbatchable(ignore_stage_f[j]),
```

Comment:
I think I intentionally had a superfluous not here, because I thought it made things clearer. But I agree with the change, this is better overall.